src/transformers/models/bert/tokenization_bert_tf.py

```diff
@@ -0,0 +1,70 @@
+from typing import List
+
+import tensorflow as tf
+
+from tensorflow_text import FastBertTokenizer, case_fold_utf8, combine_segments, pad_model_inputs
+
+from ... import PreTrainedTokenizer
+
+
+class TFBertTokenizer(tf.keras.layers.Layer):
+    # TODO Input normalization?
+    # TODO Is texts_a / texts_b the right way to handle inputs? Should it just be a multidimensional tensor?
+    # TODO Should this be a more complete class rather than reading data from an existing tokenizer?
+    # TODO Add imports and maybe some kind of AutoModel to make this findable by users
+    # TODO Do we need to change the name? Most TF users will still want the normal tokenizers
+    # TODO Add tests, particularly one with a full model and one with saving to savedmodel, as these are main use cases
+    # TODO Properly variable padding
+
+    def __init__(self, vocab_list: List, do_lower_case: bool, cls_token_id: int, sep_token_id: int, pad_token_id: int):
+        super().__init__()
+        self.tf_tokenizer = FastBertTokenizer(vocab_list, token_out_type=tf.int64)
+        self.vocab_list = vocab_list
+        self.do_lower_case = do_lower_case
+        self.cls_token_id = cls_token_id
+        self.sep_token_id = sep_token_id
+        self.pad_token_id = pad_token_id
+
+    @classmethod
+    def from_tokenizer(cls, tokenizer: PreTrainedTokenizer):
```

Comment:
If it works for both slow and fast tokenizers, the class should be "PreTrainedTokenizerBase"

Also please use forward references to avoid doing circular imports just for the sake of type-hinting.

Reply:
Fixed!